src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -3957,7 +3954,7 @@ QgsBox3D QgsPostgresProvider::extent3D() const
       sql = QStringLiteral( "SELECT %1(%2%3) FROM %4%5" )
             .arg( connectionRO()->majorVersion() < 2 ? "extent" : "ST_3DExtent",
                   quotedIdentifier( mBoundingBoxColumn ),
-                  mSpatialColType == SctPcPatch ? "::geometry" : "",
+                  ( mSpatialColType == SctPcPatch || mSpatialColType == SctGeography ) ? "::geometry" : "",
```

Comment:
cannot be equals here and diff line 3900

Reply:
Line 3957 is not inside the branch entered by line 3900.
The code goes: 
   - line 3900: if we estimate, fetch extent by estimating
   - line 3957: if ext is empty (we were unable to estimate, or we decided not to estimate), compute extent

Reply:
my bad!